python/dlr/tf_model.py

```diff
@@ -0,0 +1,153 @@
+# coding: utf-8
+import os
+import tensorflow as tf
+
+# A prefix that will be prepended to the names in graph_def
+PREFIX = "import"
+
+
+class TFModelImpl:
+    """
+    TFModelImpl is a wrapper on top of tensorflow which implements DLRModel API
+
+    Parameters
+    ----------
+    frozen_graph_file : str
+        Full path to frozen graph (.pb file)
+    dev_type : str
+        Optional. Device type ('cpu' or 'gpu')
+    dev_id : int
+        Optional. Device ID
+    """
+    def __init__(self, frozen_graph_file, dev_type=None, dev_id=None):
+        if not os.path.exists(frozen_graph_file):
+            raise ValueError("frozen_graph_file %s doesn't exist" % frozen_graph_file)
+        self.frozen_graph_file = frozen_graph_file
+
+        device = None
+        if dev_type is not None:
+            devices = ["cpu", "gpu"]
+            if dev_type not in devices:
+                raise ValueError("Invalid device type {}. Valid devices: {}".format(dev_type, devices))
+            dev_id = 0 if dev_id is None else dev_id
+            device = "/{}:{}".format(dev_type, dev_id)
+
+        self.graph = self._load_frozen_graph(frozen_graph_file, device)
+        self.input_tensor_names, self.output_tensor_names = self._get_input_and_output_names(self.graph)
+        self.input_values = {}
+
+    @staticmethod
```

Comment:
don't need static here?

Reply:
If class method does not use `self.` variables it's better to mark it as static (at least PyCharm suggests it this way).